# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1403903368 14400
#      Fri Jun 27 17:09:28 2014 -0400
# Node ID 4586051a5ff11e2ee5c1b29b171d96b1257a3295
# Parent  2b82d2f29a7b564dca8fc9a509c73aedbda88ecb
WIFEXITED and WEXITSTATUS macros for mkoctfile.cc

* mkoctfile.in.cc (WIFEXITED, WEXITSTATUS): Define for systems that
don't have these macros.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -32,16 +32,32 @@ along with Octave; see the file COPYING.
 #include <algorithm>
 #include <iostream>
 #include <fstream>
 #include <vector>
 #include <cstdlib>
 
 #include <unistd.h>
 
+// This mess suggested by the autoconf manual.
+
+#include <sys/types.h>
+
+#if defined HAVE_SYS_WAIT_H
+#include <sys/wait.h>
+#endif
+
+#ifndef WIFEXITED
+#define WIFEXITED(stat_val) (((stat_val) & 255) == 0)
+#endif
+
+#ifndef WEXITSTATUS
+#define WEXITSTATUS(stat_val) (static_cast<unsigned> (stat_val) >> 8)
+#endif
+
 static std::map<std::string, std::string> vars;
 
 #ifndef OCTAVE_VERSION
 #define OCTAVE_VERSION %OCTAVE_CONF_VERSION%
 #endif
 
 #ifndef OCTAVE_PREFIX
 #define OCTAVE_PREFIX %OCTAVE_CONF_PREFIX%
@@ -339,19 +355,22 @@ ends_with (const std::string& s, const s
           && s.rfind (suffix) == s.length () - suffix.length ());
 }
 
 static int
 run_command (const std::string& cmd)
 {
   if (debug)
     std::cout << cmd << std::endl;
+
   int result = system (cmd.c_str ());
+
   if (WIFEXITED (result))
     result = WEXITSTATUS (result);
+
   return result;
 }
 
 bool
 is_true (const std::string& s)
 {
   return (s == "yes" || s == "true");
 }
